lib/module.js

```diff
@@ -109,14 +110,15 @@ function tryPackage(requestPath, exts, isMain) {
 }
 
 // check if the file exists and is not a directory
-// resolve to the absolute realpath if running main module,
-// otherwise resolve to absolute while keeping symlinks intact.
+// if using --preserve-symlinks and isMain is false,
+// keep symlinks intact, otherwise resolve to the
+// absolute realpath.
 function tryFile(requestPath, isMain) {
   const rc = stat(requestPath);
-  if (isMain) {
-    return rc === 0 && fs.realpathSync(requestPath);
+  if (preserveSymlinks && !isMain) {
```

Comment:
Yes, I know. I switched it to this purposefully because it flows a bit more
logically. Specifically, we only care about the isMain check if preserve
symlinks is on. If it's off, there's no reason to check isMain.
On May 12, 2016 3:13 AM, "Ben Noordhuis" <notifications@github.com> wrote:

In lib/module.js
<https://github.com/nodejs/node/pull/6537#discussion_r62996232>:

>  function tryFile(requestPath, isMain) {
>    const rc = stat(requestPath);
> -  if (isMain) {
> -    return rc === 0 && fs.realpathSync(requestPath);
> +  if (preserveSymlinks && !isMain) {

The diff could be made smaller if you write this as if (isMain &&
!preserveSymlinks) {.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/nodejs/node/pull/6537/files/ad138be3ec882a3370cd8ab1b1902264b8c2ccda#r62996232>